### PR TITLE
Introduce quantized convolution serialization format 3

### PR DESCRIPTION
**Pull request by dreiss**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #60242 Swich qconv serialization to format version 3
* **#60241 Introduce quantized convolution serialization format 3**
* #60576 Remove unnecessary temporary fix in quant bc test
* #60240 Fix submodule naming in ONNX quantized test

Summary:
We're going to make a forward-incompatible change to this serialization
format soon, so I'm taking the opportunity to do a little cleanup.

- Use int for version.  This was apparently not possible when V2
  was introduced, but it works fine now as long as we use int64_t.
  (Note that the 64-bits are only used in memory.  The serializer will
  use 1 byte for small non-negative ints.)
- Remove the "packed params" tensor and replace it with a list of ints.
- Replace the "transpose" field with "flags" to allow more binary flags
  to be packed in.
- Unify required and optional tensors.  I just made them all optional
  and added an explicit assertion for the one we require.

A bit of a hack: I added an always-absent tensor to the front of the
tensor list.  Without this, when passing unpacked params from Python to
the ONNX JIT pass, they type would be inferred to `List[Tensor]` if all
tensors were present, making it impossible to cast to
`std::vector<c10::optional<at:Tensor>>` without jumping through hoops.

The plan is to ship this, along with another diff that adds a flag to
indicate numerical requirements, wait a few weeks for an FC grace
period, then flip the serialization version.

Test Plan:
CI.  BC tests.

### CODE REVIEW

File: aten/src/ATen/native/quantized/cpu/conv_serialization.h [hunk 1]
```diff
@@ -37,23 +39,45 @@
  *  2. list of optional tensors
  *    0: bias
  *
- *  Note: version is a string and conv params are packed into a Tensor
- *    to make ONNX happy (ints and containers of ints are not supported).
+ * Version 3
+ *
+ * - Fields:
+ *  0. version (int)
+ *  1. list of integer configuration values
```
Comment (vkuzo) [inline aten/src/ATen/native/quantized/cpu/conv_serialization.h:46]: `int64_t`?

Comment (dreiss) [reply aten/src/ATen/native/quantized/cpu/conv_serialization.h:46]: Looks like this breaks in Visual C++ . :/

**Comment by dreiss** [comment]
This also reduce model size of quantized mobilenetv2 by ~10kB.  Probably even more once we enable compression for constants.pkl.

**Review by vkuzo — APPROVED** [review]

**Comment by vkuzo** [comment]
this is great!

**Pull request closed without merging**